Tasks/Common/azure-blobstorage-artifactProvider/azureBlobStorageProvider.ts

```diff
@@ -30,8 +29,8 @@ export class AzureBlobProvider implements models.IArtifactProvider {
                 } else {
                     var blobUrl = self._blobSvc.getUrl(self._container, blobPath);
                     console.log(tl.loc("CreatedBlobForItem", item.path, blobUrl));
-                    newArtifactItem.metadata["downloadUrl"] = blobUrl;
-                    resolve(newArtifactItem);
+                    item.metadata["destinationUrl"] = blobUrl;
+                    resolve(item);
```

Comment:
You are resolving here; but wont the upload complete much later. Or is that handled correctly

Reply:
I am not sure of the intended return type: Should it Should we wait for the pipe to finish before returning the copied URL.

Reply:
callback is called only after writeStream's 'end' event.